src/sentry/sentry_metrics/use_case_id_registry.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from typing import Any, Iterator, Mapping, Optional, Set
+
+from sentry.sentry_metrics.configuration import UseCaseKey
+
+_REGISTERED_USE_CASES: dict[str, str] = {
+    "PERFORMANCE": "performance",
+    "RELEASE_HEALTH": "release-health",
+}
+
+
+class _UseCaseID(type):
+    def __getattr__(self, attr: str) -> UseCaseID:
+        if attr not in _REGISTERED_USE_CASES:
+            raise AttributeError(attr)
+
+        return UseCaseID(attr.lower())
+
+    def __iter__(self) -> Iterator[UseCaseID]:
+        return iter(
+            UseCaseID(value)
+            for value in {
+                **_REGISTERED_USE_CASES,
+            }.values()
+        )
+
+
+class UseCaseID(metaclass=_UseCaseID):
+    def __init__(self, value: str):
+        if value not in _REGISTERED_USE_CASES.values():
+            raise ValueError("Passed use case has not been registered")
+
+        self.value = value
+
+    def __hash__(self) -> int:
+        return hash(self.value)
+
+    def __eq__(self, other: Any) -> bool:
+        return isinstance(other, UseCaseID) and other.value == self.value
+
+    def __repr__(self) -> str:
+        return f"UseCaseID.{self.value.upper()}"
+
+
+def register_use_case(key: str) -> UseCaseID:
+    _REGISTERED_USE_CASES[key.upper()] = key.lower()
+    return UseCaseID(key)
+
+
+METRIC_PATH_MAPPING: Mapping[UseCaseKey, Set[UseCaseID]] = {
+    UseCaseKey.RELEASE_HEALTH: {UseCaseID.RELEASE_HEALTH},
+    UseCaseKey.PERFORMANCE: {UseCaseID.PERFORMANCE},
+}
+
+
+def get_metric_path_from_usecase(use_case: UseCaseID) -> Optional[UseCaseKey]:
```

Comment:
Why do we need the separate UseCaseID and UseCaseKey?

Reply:
So `UseCaseKey` is currently used for both configuration (see MetricsIngestConfiguration), as well as the use_case_id that is used to populate message payloads and is used in the indexer interface (e.g. the `resolve` function being used to resolve (string, use_case_id=performance) to an integer). We're going to make it so that `UseCaseKey` is solely used for ingest configuration (so, it can be either metrics or generic metrics), and that `UseCaseID` becomes the representation for all use cases (performance, release health, escalating issues, etc.) within one of those metric paths. 

We're going to rename `UseCaseKey` to `MetricPathKey` once we switch `UseCaseKey` to `UseCaseID` in all the places where `UseCaseKey` is being used to refer to these individual use cases.

Reply:
I see. Should the metric path mapping be `Mapping[UseCaseID, UseCaseKey]` then instead of `Mapping[UseCaseKey, Set[UseCaseID]]`? It seems like this function would be simpler and also it removes the possibility of invalid data like putting the same use case ID into both keys, etc

Reply:
Sure, that makes sense to me

Reply:
So, this then means that the UseCaseID is now globally unique and not namespaced by the `UseCaseKey`/`MetricPathKey` like it was before... Is that right?

Reply:
Could you clarify what you mean by "namespaced by" here? If I understood correctly, I think the answer to your question is yes.

Reply:
Previously the same use case ID could be present under both metrics paths. Now seems like it cannot. Just want to clarify the desired behavior.